step1/src/main.js

```diff
@@ -1 +1,179 @@
 // 코드 작성
+
+/**  1. 이벤트에 대한 함수 추가 **/
+// (1) 태그 추가 함수
+function appendItem (event) {
+    
+    event.preventDefault(); // handAppendItemButton
+
+    const $appender = $appenderForm.querySelector('input');
+
+    const $newItem = document.createElement('li');
+    
+    // 태그에 내용 채우기
+    if($appender.value){
+        $newItem.innerHTML = `
+        <input class="check" type="checkbox" style="margin-right:15px">
+        <p>
+            ${$appender.value}
+        </p>
+        <button class="complete" type="button">완료</button>
+        <button class="update" type="button">수정</button>
+        <button class="remove" type="button">삭제</button>
+        `;
+        $todoList.appendChild($newItem);
+        $appender.value = "";
+        $appender.focus();
+
+        // 추가한 아이템 버튼 이벤트 등록
+        /* document.addEventListener('click', newItemEvent);
+        function newItemEvent (event) {
+            switch (event.target.getAttribute("class")){
+                case "complete":
+                    toggleItem(event);
+                    break;
+                case "check":
+                    toggleItem(event);
+                    break;
+                case "update":
+                    editItem(event);
+                    break;
+                case "remove":
+                    removeItem(event);
+                    break;
+                default: break;
+            }
+        } */
+
+        $newItem.querySelector('.remove').onclick = removeItem;
+        $newItem.querySelector('.complete').onclick = toggleItem;
+        $newItem.querySelector('.check').onclick = toggleItem;
+        $newItem.querySelector('.update').onclick = editItem;
+    }
+    else{alert("아이템 이름을 입력해주세요.")};
```

Comment:
```suggestion
    if(!$appender.value) {
      return alert('아이템 이름을 입력해주세요!');
    }
    $newItem.innerHTML = `
      <input class="check" type="checkbox" style="margin-right:15px">
      <p>
          ${$appender.value}
      </p>
      <button class="complete" type="button">완료</button>
      <button class="update" type="button">수정</button>
      <button class="remove" type="button">삭제</button>
    `;
    $todoList.appendChild($newItem);
    $appender.value = "";
    $appender.focus();
    $newItem.querySelector('.remove').onclick = removeItem;
    $newItem.querySelector('.complete').onclick = toggleItem;
    $newItem.querySelector('.check').onclick = toggleItem;
    $newItem.querySelector('.update').onclick = editItem;
```
이렇게 분기문 내부로 진입하는 코드를 최대한 없애주세요!